File: Dna_pairs.java
public class Dna_pairs {
	public static void main(String[] args){
		String strand_1 = "AATGCCTATGGC";
		String pair;
		for(int i = 0; i < strand_1.length(); i++){
			char c = strand_1.charAt(i);
			String C = Character.toString(c);

			if(C.equals("A"))
			{
				System.out.println("A T");
			}
			else if(C.equals("T"))
			{
				System.out.println("T A");
			}
			else if(C.equals("G"))
			{
				System.out.println("G C");
			}
			else if (C.equals("C"))
			{
				System.out.println("C G");
			}
			else
			{
				System.out.println("ERROR");
			}
		}

	}
}

File: Dna_pairs.scala
import scala.collection.JavaConversions._

object Dna_pairs {

  def main(args: Array[String]) {
    val strand_1 = "AATGCCTATGGC"
    val pair: String = null
    for (i <- 0 until strand_1.length) {
      val c = strand_1.charAt(i)
      val C = java.lang.Character.toString(c)
      if (C == "A") {
        println("A T")
      } else if (C == "T") {
        println("T A")
      } else if (C == "G") {
        println("G C")
      } else if (C == "C") {
        println("C G")
      } else {
        println("ERROR")
      }
    }
  }
}
